src/v/serde/parquet/encoding.cc

```diff
@@ -19,78 +19,78 @@
 
 namespace serde::parquet {
 
-iobuf encode_plain(const chunked_vector<boolean_value>& vals) {
-    iobuf buf;
-    size_t i = 0;
-    while (i < vals.size()) {
-        size_t pass_size = std::min<size_t>(vals.size() - i, CHAR_BIT);
-        uint8_t tmp = 0;
-        size_t shift = 0;
-        for (; shift < pass_size; ++i, ++shift) {
-            if (vals[i].val) {
-                tmp |= 1U << shift;
-            }
-        }
-        buf.append(&tmp, 1);
+void plain_encoder<boolean_value>::add_value(boolean_value v) {
+    bits |= static_cast<uint8_t>(v.val) << shift;
+    shift = ++shift % CHAR_BIT;
+    if (shift == 0) {
+        buf.append(&bits, 1);
+        bits = 0;
     }
-    return buf;
 }
 
-iobuf encode_plain(const chunked_vector<int32_value>& vals) {
-    iobuf buf;
-    for (const auto& v : vals) {
-        int32_t i = ss::cpu_to_le(v.val);
-        // NOLINTNEXTLINE(*reinterpret-cast*)
-        buf.append(reinterpret_cast<const uint8_t*>(&i), sizeof(int32_t));
+iobuf plain_encoder<boolean_value>::get_encoded_buf() {
+    if (shift != 0) {
+        buf.append(&bits, 1);
+        bits = 0;
+        shift = 0;
     }
-    return buf;
+    return std::exchange(buf, {});
 }
 
-iobuf encode_plain(const chunked_vector<int64_value>& vals) {
-    iobuf buf;
-    for (const auto& v : vals) {
-        int64_t i = ss::cpu_to_le(v.val);
-        // NOLINTNEXTLINE(*reinterpret-cast*)
-        buf.append(reinterpret_cast<const uint8_t*>(&i), sizeof(int64_t));
-    }
-    return buf;
+size_t plain_encoder<boolean_value>::size_bytes() const {
+    return sizeof(*this) + buf.size_bytes();
 }
 
-iobuf encode_plain(const chunked_vector<float32_value>& vals) {
-    iobuf buf;
-    for (const auto& v : vals) {
-        // NOLINTNEXTLINE(*reinterpret-cast*)
-        buf.append(reinterpret_cast<const uint8_t*>(&v.val), sizeof(float));
+template<typename value_type, bool conv_le>
+void numeric_plain_encoder<value_type, conv_le>::add_value(value_type v) {
+    if constexpr (conv_le) {
+        v.val = ss::cpu_to_le(v.val);
```

Comment:
optional consider removing the second template param and using `!std::is_floating_point<decltype(v.val)>` or `std::is_integral<decltype(v.val)>`?